src/aiida/engine/processes/calcjobs/calcjob.py

```diff
@@ -974,6 +974,14 @@ def presubmit(self, folder: Folder) -> CalcInfo:
             codes_run_mode = calc_info.codes_run_mode
 
         job_tmpl.codes_run_mode = codes_run_mode
+
+        if calc_info.file_copy_order is not None:
```

Comment:
I assume these lines are needed because the `DefaultFieldsAttributeDict` class doesn't allow for validation or specifying defaults? Are we also considering using `pydantic` for the `CalcInfo` class? ^^

Reply:
Indeed. If we can switch safely without breaking backwards compatibility, we definitely could. But these classes are so widely used that we should be very careful